### PR TITLE
GlobalDNS alidns provider

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/17838

Related types PR: https://github.com/rancher/types/pull/706

1. Alidns provider [use configs from a config file](https://github.com/kubernetes-incubator/external-dns/blob/master/provider/alibaba_cloud.go#L101), otherwise it fetches required infos from metadata limiting the external-dns to run on a alibaba cloud node with necessary RAM roles. Therefore a secret is created to mount the config file.
2. System timezone directory is mounted to the pod, it is used by Ali cloud sdk.

### CODE REVIEW

File: pkg/controllers/management/globaldns/alidns_template.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: {{.providerName}}
+    spec:
+      serviceAccountName: external-dns
+      containers:
+      - name: {{.providerName}}
+        image: registry.opensource.zalan.do/teapot/external-dns:latest
```
Comment (orangedeng) [inline pkg/controllers/management/globaldns/alidns_template.go]: Should this image be in rancher repo? Incase we can not pull it.

File: pkg/controllers/management/globaldns/globaldns_provider.go [hunk 2]
```diff
@@ -247,3 +262,43 @@ func (n *ProviderLauncher) createExternalDNSDeployment(obj *v3.GlobalDNSProvider
...
+		Name:       obj.Name,
+		APIVersion: "v3",
+		UID:        obj.UID,
+		Kind:       obj.Kind,
+		Controller: &controller,
+	}}
+	secret.ObjectMeta.OwnerReferences = ownerRef
+
+	secretCreated, err := n.Secrets.Create(secret)
+	if err != nil {
```
Comment (orangedeng) [inline pkg/controllers/management/globaldns/globaldns_provider.go]: Should we check if the secret already exists?

File: pkg/controllers/management/globaldns/alidns_template.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        args:
+        - --source=service
+        - --source=ingress
+        - --domain-filter={{.rootDomain}}
+        - --provider=alibabacloud
+        - --alibaba-cloud-zone-type=public # only look at public hosted zones (valid values are public, private or no value for both)
+        - --registry=txt
+        - --txt-owner-id=my-identifier
+        - --alibaba-cloud-config-file=/etc/alibaba/config.yaml
```
[the remark this answers is not in the document]
Comment (gitlawr) [reply pkg/controllers/management/globaldns/alidns_template.go]: We can, but it is strange to use this default value while the config file is parsed as a yaml file [here](https://github.com/kubernetes-incubator/external-dns/blob/master/provider/alibaba_cloud.go#L106).

**Comment by orangedeng** [comment]
This PR rename the external-dns image from `registry.opensource.zalan.do/teapot/external-dns:latest` to `rancher/kubernetes-external-dns`. We need to push that image to rancher repo.

**Comment by prachidamle** [comment]
LGTM, merge after types

**Pull request merged**